gst/tensor_mux/gsttensormux.c

```diff
@@ -358,38 +402,61 @@ gst_tensor_mux_collect_buffer (GstTensorMux * tensor_mux,
     walk = g_slist_next (walk);
 
     GstBuffer *buf = NULL;
-    buf = gst_collect_pads_pop (tensor_mux->collect, data);
 
-    if (GST_IS_BUFFER (buf)) {
-      if (GST_BUFFER_PTS_IS_VALID (buf)) {
-        pad->pts_timestamp =
-            gst_segment_to_running_time (&data->segment, GST_FORMAT_TIME,
-            GST_BUFFER_PTS (buf));
+    if (tensor_mux->synch) {
+      buf = gst_collect_pads_peek (tensor_mux->collect, data);
+      if (buf == NULL && pad->buffer == NULL)
+        return TRUE;
+
+      if (buf != NULL) {
+        if (GST_BUFFER_PTS (buf) < tensor_mux->current_time) {
+          gst_buffer_unref (buf);
+          if (pad->buffer != NULL)
+            gst_buffer_unref (pad->buffer);
+          pad->buffer = gst_collect_pads_pop (tensor_mux->collect, data);
+          silent_debug ("Fame Dropped (# %d) : %lu", counting,
+              GST_BUFFER_PTS (buf));
```

Comment:
line 418  GST_BUFFER_PTS (buf)); 
buf unref in line 413
Please change  buf to pad->buffer, or move this log before buffer_unref.